core/lib/multivm/src/versions/vm_fast/vm.rs

```diff
@@ -637,6 +648,11 @@ impl<S: ReadStorage, Tr: Tracer + Default + 'static> VmInterface for Vm<S, Tr> {
         let gas_remaining = self.gas_remaining();
         let gas_used = gas_before - gas_remaining;
 
+        // We need to filter out bytecodes the deployment of which may have been reverted; the tracer is not aware of reverts.
+        // To do this, we check bytecodes against deployer events.
+        let factory_deps_marked_as_known = extract_bytecodes_marked_as_known(&logs.events);
+        let new_known_factory_deps = self.world.decommit_bytecodes(&factory_deps_marked_as_known);
```

Comment:
Is it possible to include only EVM bytecodes here? IIUC there is no point in storing era-vm code in `new_known_factory_deps `.

Reply:
There's a TODO for this; I'd prefer to work on it separately (looks quite easy to do, maybe deceptively so).

https://github.com/matter-labs/zksync-era/blob/8ae06b237647715937fb3656d881c0fd460f2a07/core/lib/vm_interface/src/types/outputs/execution_result.rs#L124-L125